src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -229,15 +229,21 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude) || mode == Mode.AppendVersion)
+            if (!string.IsNullOrEmpty(Href) && AppendVersion == true)
             {
-                // No globbing to do, just build a <link /> tag to match the original one in the source file.
-                // Or just add file version to the link tag.
-                BuildLinkTag(attributes, builder);
+                EnsureFileVersionProvider();
+                output.Attributes[HrefAttributeName].Value = _fileVersionProvider.AddFileVersionToPath(Href);
```

Comment:
In this case a slight change is needed since Href can be null. 
`output.Attributes[HrefAttributeName].Value as string` =>
`output.Attributes[HrefAttributeName]?.Value as string`.